### PR TITLE
CL-78 Privacy Flags

### PR DESCRIPTION
**Pull request by stevelw**

### CODE REVIEW

File: __tests__/app.test.ts [hunk 1]
```diff
@@ -101,6 +101,30 @@ describe('🧪 Express Application', () => {
...
 			});
 		});
+
+		describe('PATCH /api/users/:id', () => {
+			it('204: should return no content on successful update', () => {
+				const data = {
+					requestPrivacy: true,
+				};
+				return request(app)
+					.patch('/api/users/cm3op7iwu0000jrcqa60tc9kv')
```
Comment (guarde) [inline __tests__/app.test.ts]: Should the route be /users/:id? Could we not do /api/user/settings and have /user relate to the currently logged in user?

File: src/controllers/users.ts [hunk 2]
```diff
@@ -21,3 +23,23 @@ export function getCats(request: Request, response: Response) {
...
+	if (!result.success) return next({ status: 400, message: result.errors });
+
+	const { id } = req.params;
+	users
+		.updateUser(id, req.body)
+		.then((updatedUser) => {
+			res.status(204).json({ sucess: true, data: updatedUser });
+		})
+		.catch(() => {
+			res.status(500).send({ msg: 'Server error' });
```
Comment (guarde) [inline src/controllers/users.ts]: Could we change this message to fit the rest - It's typically 'An internal server error occurred'

File: src/models/users.ts [hunk 3]
```diff
@@ -7,3 +7,17 @@ export function getUserByUsername(username: string) {
...
+export function updateUser(
+	userId: string,
+	data: { requestedPrivacy: boolean }
+) {
+	return extendedClient.user.update({
+		where: {
+			id: userId,
+		},
+		data: {
+			requested_privacy: data.requestedPrivacy,
```
Comment (guarde) [inline src/models/users.ts]: If we replace this with {...data}, you can make this function more reusable by adding optional parameters in data

File: __tests__/app.test.ts (hunk 1 above)
Comment (stevelw) [reply __tests__/app.test.ts]: Done this, but used user**s** as it should relate to the table it updates.

File: src/controllers/users.ts [hunk 4]
```diff
@@ -21,3 +23,24 @@ export function getCats(request: Request, response: Response) {
...
+export function updateUser(req: Request, res: Response, next: NextFunction) {
+	const schema = {
+		requested_privacy: 'boolean',
+	};
+	const payload = req.body;
+	const result = validator(payload, schema);
+
+	if (!result.success) return next({ status: 400, message: result.errors });
+
+	const id: string = req.headers.authorization ?? '';
```
Comment (guarde) [inline src/controllers/users.ts]: You can just use the middleware I made :) Takes care of everything

**Review by guarde — CHANGES_REQUESTED** [review]

File: src/controllers/users.ts (hunk 4 above)
Comment (stevelw) [reply src/controllers/users.ts]: How?

**Comment by stevelw** [comment]
Just changing from boolean to string to allow the multiple options in our plan

File: src/routes/users.ts [hunk 5]
```diff
@@ -5,5 +5,6 @@ const users: Router = Router();
 
 users.get('/:id/devices', controller.getDevices);
 users.get('/:id/cats', controller.getCats);
+users.patch('/settings', controller.updateUser);
```
Comment (guarde) [inline src/routes/users.ts]: You're still not using the auth middleware

**Review by guarde — CHANGES_REQUESTED** [review]

File: src/routes/users.ts (hunk 5 above)
Comment (stevelw) [reply src/routes/users.ts]: You never replied to my comment about that above.

Comment (guarde) [reply src/routes/users.ts]: In the route, add the middleware [like this](https://github.com/clowder-cloud/backend/blob/1b24f8574f26875a9a83c69215e79889bf64cb16/src/routes/cats.ts#L7). You can then replace the first parameter of your controller to the user, [like this](https://github.com/clowder-cloud/backend/blob/1b24f8574f26875a9a83c69215e79889bf64cb16/src/controllers/cats.ts#L7)

**Review by guarde — APPROVED** [review]

**Pull request merged**